spec/commands/add_spec.rb

```diff
@@ -105,4 +105,16 @@
     bundle "add 'baz' --source='file://does/not/exist'"
     expect(out).to include("Could not fetch specs from file://does/not/exist/")
   end
+
+  it "throws an error if a gem is added which is already specified in Gemfile" do
+    install_gemfile <<-G
+      source "file://#{gem_repo2}"
+      gem "rack", "1.0"
+    G
+
+    bundle "add 'rack'"
+
+    expect(out).to include "You cannot specify the same gem twice with different version requirements"
+    expect(out).to include "If you want to update the gem version, run `bundle update rack`. You may need to change the version requirement specified in the Gemfile if it's too restrictive"
```

Comment:
Interesting - this test example highlights that this is a poor error message for this particular case. As a user, if I did ran this (`bundle add rake`), I'd expect to see an error that says "gem already added. Skip adding the gem" and not the above message. I can see this being a potential for confusion in future.

I'd only expect the above error (i.e. cannot specify same gem twice with different versions) when I tried to run `bundle add rack --version "1.4"`.

Would it be simple for you to make this distinction in error messages between these 2 cases?

Reply:
Done. Please have a look :smile: